python/ray/train/lightgbm/config.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from dataclasses import dataclass
+
+import ray
+from ray.train._internal.utils import get_address_and_port
+from ray.train._internal.worker_group import WorkerGroup
+from ray.train.backend import Backend, BackendConfig
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class LightGBMConfig(BackendConfig):
+    """Configuration for LightGBM distributed data-parallel training setup.
+
+    Ray Train will set up the necessary coordinator processes and environment
+    variables for your workers to communicate with each other.
+
+    See the LightGBM docs for more information on the "network parameters"
+    that Ray Train sets up for you:
+    https://lightgbm.readthedocs.io/en/latest/Parameters.html#network-parameters
+    """
+
+    NETWORK_PARAMS_KEY = "LIGHTGBM_NETWORK_PARAMS"
```

Comment:
nit: Any reason this is defined in the Config rather than globally somewhere?

Reply:
No reason in particular 😄  do you prefer just as a constant in this file?

Reply:
I guess it depends how much this ends up being exposed to users. I see there's an assertion that mentions it, but the dependency between the Config <> Backend <> constant might be confusing for users.